The JSON import is not as expected.

<!-- (At least include the following, feel free to add if you have more content) -->
### Steps to reproduce the behavior (Required)
1. create table.
```SQL
CREATE TABLE `primary_table_without_null` (
    `k1` date NOT NULL, `k2` datetime NOT NULL, `k3` string NOT NULL, `k4` string NOT NULL, `k5` boolean NOT NULL, 
    `v1` tinyint NOT NULL, `v2` smallint NOT NULL, `v3` int NOT NULL, `v4` bigint NOT NULL, `v5` largeint NOT NULL, `v6` float NOT NULL, `v7` double NOT NULL, `v8` decimal(27,9) NOT NULL 
) ENGINE=OLAP 
PRIMARY KEY(`k1`, `k2`, `k3`, `k4`, `k5`) 
COMMENT "OLAP"
DISTRIBUTED BY HASH(`k1`, `k2`, `k3`, `k4`, `k5`) 
BUCKETS 3 PROPERTIES ( "replication_num" = "1", "storage_format" = "v2" );
```
2. stream load.
curl --location-trusted -u root:
    -T /basic_types_data.json -XPUT -H label:stream_load_1646904047656 -H format:json
    -H columns:k1, k2, k3, k4, k5, v1, v2, v3, v4, v5, v6, v7, v8
    -H jsonpaths:["$.k1", "$.k2", "$.k3", "$.k4", "$.k5", "$.k6", "$.k7", "$.k8", "$.k9", "$.k10", "$.k11", "$.k12", "$.k13", "$.k13"]
    http://x.x.x.x:8234/api/stream_load_test_db_1646904047518/primary_table_without_null/_stream_load

data file:
```
{
    "k1": "2020-06-23",
    "k2": "2020-06-23 00:00:09",
    "k3": "beijing",
    "k4": "haidian",
    "k5": 0,
    "k6": -119,
    "k7": -32759,
    "k8": -2147483639,
    "k9": -9223372036854775799,
    "k10": "-18446744073709551607",
    "k11": -2.2,
    "k12": -3.05,
    "k13": -3.132
}
```
### Expected behavior (Required)
import ok.
### Real behavior (Required)
```AssertionError: too many filtered rows```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - 2.0 2.1

--- a/be/src/exec/vectorized/json_scanner.cpp
+++ b/be/src/exec/vectorized/json_scanner.cpp
@@ -325,15 +325,19 @@ Status JsonReader::read_chunk(Chunk* chunk, int32_t rows_to_read, const std::vec
                     }
                     rapidjson::Value* json_values = JsonFunctions::get_json_object_from_parsed_json(
                             _scanner->_json_paths[i], objectValue, _origin_json_doc.GetAllocator());
-                    if (json_values == nullptr) {
-                        if (strcmp(column_name, "__op") == 0) {
-                            // special treatment for __op column, fill default value '0' rather than null
-                            column->append_strings(literal_0_slice_vector);
+                    if (strcmp(column_name, "__op") == 0) {
+                        // special treatment for __op column, fill default value '0' rather than null
+                        if (json_values != nullptr && json_values->IsInt() && json_values->GetInt() == 1) {
+                            column->append_strings(literal_1_slice_vector);
                         } else {
-                            column->append_nulls(1);
+                            column->append_strings(literal_0_slice_vector);
                         }
                     } else {
-                        _construct_column(*json_values, column.get(), slot_descs[i]->type());
+                        if (json_values == nullptr) {
+                            column->append_nulls(1);
+                        } else {
+                            _construct_column(*json_values, column.get(), slot_descs[i]->type());
+                        }
                     }
                 }
             }

